File: PhosphOps/src/main/java/tn/esprit/spring/phosphops/Service/EquipementService.java
package tn.esprit.spring.phosphops.Service;

import tn.esprit.spring.phosphops.DTO.EquipementDTO;

import java.util.List;

public interface EquipementService {
    EquipementDTO create(EquipementDTO dto);
    EquipementDTO update(Long id, EquipementDTO dto);
    EquipementDTO getById(Long id);
    List<EquipementDTO> getAll();
    void delete(Long id);
}

File: PhosphOpsFront/src/app/services/equipement.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { Equipement } from '../models/equipement.model';

@Injectable({
  providedIn: 'root'
})
export class EquipementService {

private readonly apiUrl = 'https://phosphopsbackend-bac3czavgtbea6aw.swedencentral-01.azurewebsites.net/PhosphOps/api/equipements';
  constructor(private http: HttpClient) {}

  getAll(): Observable<Equipement[]> {
    return this.http.get<Equipement[]>(this.apiUrl);
  }

  getById(id: number): Observable<Equipement> {
    return this.http.get<Equipement>(`${this.apiUrl}/${id}`);
  }

  create(equipement: Equipement): Observable<Equipement> {
    return this.http.post<Equipement>(this.apiUrl, equipement);
  }

  update(id: number, equipement: Equipement): Observable<Equipement> {
    return this.http.put<Equipement>(`${this.apiUrl}/${id}`, equipement);
  }

  delete(id: number): Observable<void> {
    return this.http.delete<void>(`${this.apiUrl}/${id}`);
  }
}